src/main/java/org/opentripplanner/graph_builder/module/geometry/GeometryAndBlockProcessor.java

```diff
@@ -65,11 +62,11 @@
 
     private OtpTransitService transitService;
 
-    private Map<ShapeSegmentKey, LineString> geometriesByShapeSegmentKey = new HashMap<ShapeSegmentKey, LineString>();
+    private Map<ShapeSegmentKey, LineString> geometriesByShapeSegmentKey = new ConcurrentHashMap<>();
 
-    private Map<FeedScopedId, LineString> geometriesByShapeId = new HashMap<FeedScopedId, LineString>();
+    private Map<FeedScopedId, LineString> geometriesByShapeId = new ConcurrentHashMap<>();
 
-    private Map<FeedScopedId, double[]> distancesByShapeId = new HashMap<>();
+    private Map<FeedScopedId, double[]> distancesByShapeId = new ConcurrentHashMap<>();
```

Comment:
We discussed these on todays developer meeting. The conclusion is this is ok, but that we should comment the thread-safety for this class int the class JavaDoc -> Only these three maps are thread-safe and can be updated in a concurrent context.